ovos_config/base.py

```diff
@@ -0,0 +1,272 @@
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import re
+from os.path import exists, isfile
+
+import yaml
+
+from ovos_config.locations import USER_CONFIG, SYSTEM_CONFIG, WEB_CONFIG_CACHE, DEFAULT_CONFIG
+from ovos_utils import camel_case_split
+from ovos_utils.json_helper import load_commented_json, merge_dict
+from ovos_utils.log import LOG
+
+try:
```

Comment:
Would it make sense to put remote stuff in its own submodule?

Reply:
depending on what you mean by remote stuff, https://github.com/OpenVoiceOS/selene_api is not yet ready but will be used here

i thought about adding a remote.py in this package but it would only contain the RemoteConfig class which looks right in base.py together with equivalent classes